packages/loader/container-loader/src/containerStorageAdapter.ts

```diff
@@ -109,15 +109,20 @@ export class ContainerStorageAdapter
 			this.logger,
 		));
 
-		this._summarizeProtocolTree =
-			this._summarizeProtocolTree ?? service.policies?.summarizeProtocolTree;
-		if (this.summarizeProtocolTree) {
-			this.logger.sendTelemetryEvent({ eventName: "summarizeProtocolTreeEnabled" });
-			this._storageService = new ProtocolTreeStorageService(
-				retriableStorage,
-				this.addProtocolSummaryIfMissing,
-			);
-		}
+		// A storage service wrapper which intercept calls to uploadSummaryWithContext and ensure they include
+		// the protocol summary, provided single-commit summary is enabled.
+		this._storageService = new ProtocolTreeStorageService(
+			retriableStorage,
```

Comment:
It looks like a bug (that your code is fixing), but worth doing some more digging to make sure we are not missing something (and not changing behavior that we do not understand):
The code before did not use `retriableStorage` if this.summarizeProtocolTree === false.
Now we use it all the time.

It looks correct, but raising to make sure we do proper due dilligence here.

Reply:
`retriableStorage` was being used before as well. Line 107 above is a double assignment statement. Easy to overlook :D